### PR TITLE
pool: implemented Role Stringer interface

### PR DESCRIPTION
**Pull request by KaymeKaydex**
implemented Role Stringer interface for human-readable printing

Closes: #405

### CODE REVIEW

**Comment by oleg-jukovec** [comment]
I suggest to use `stringer` + `go generate` instead of a custom code.
```
diff --git a/pool/const.go b/pool/const.go
index 7ec239f..22bff4f 100644
--- a/pool/const.go
+++ b/pool/const.go
@@ -30,8 +30,12 @@ const (
 // Role describes a role of an instance by its mode.
 type Role uint32
 
+//go:generate stringer -type Role -linecomment
 const (
-    UnknownRole Role = iota // A connection pool failed to discover mode of the instance.
-    MasterRole              // The instance is read-write mode.
-    ReplicaRole             // The instance is in read-only mode.
+    // A connection pool failed to discover mode of the instance.
+    UnknownRole Role = iota // unknown
+    // The instance is read-write mode.
+    MasterRole              // master
+    // The instance is in read-only mode.
+    ReplicaRole             // replica
 )
 ```

File: pool/const_test.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// TestRole_String checking that print is non-number(len = 1 ) Role
+func TestRole_String(t *testing.T) {
+	require.True(t, len(fmt.Sprintf("%v", UnknownRole)) > 1)
+	require.True(t, len(fmt.Sprintf("%v", MasterRole)) > 1)
+	require.True(t, len(fmt.Sprintf("%v", ReplicaRole)) > 1)
```
Comment (oleg-jukovec) [inline pool/const_test.go:14]: Let's compare real values ​​instead of the existence check.

**Comment by KaymeKaydex** [comment]
@oleg-jukovec i generated Stringer face by stringer library. since it can only cut prefixes, I was forced to rename the roles, since they are not specified externally, this will not change the user interface

**Comment by oleg-jukovec** [comment]
> @oleg-jukovec i generated Stringer face by stringer library. since it can only cut prefixes, I was forced to rename the roles, since they are not specified externally, this will not change the user interface

Unfortunately, we could not change naming because it will break backward compatibility.

But the ‘stringer’ could generate custom strings, see ‘-linecoment’ option for the ‘stringer’ and my patch more carefully.

File: pool/const_test.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestRole_String(t *testing.T) {
+	require.Equal(t, "unknown", UnknownRole)
+	require.Equal(t, "master", MasterRole)
+	require.Equal(t, "replica", ReplicaRole)
```
Comment (oleg-jukovec) [inline pool/const_test.go:12]:
```suggestion
	require.Equal(t, "unknown", UnknownRole.String())
	require.Equal(t, "master", MasterRole.String())
	require.Equal(t, "replica", ReplicaRole.String())
```

**Review by oleg-jukovec — CHANGES_REQUESTED** [review]
Please, fix red tests.

**Review by oleg-jukovec — APPROVED** [review]

**Review by DerekBum — COMMENTED** [review]
Thank you for the patch! Please see couple of nits below.

Also, don't forget to squash the commits after the work is done. And please change the line `Related issues: #405` to the `Closes #405`, so the issue could be closed automatically after the merge.

**Comment by KaymeKaydex** [comment]
@DerekBum thanks, I accepted the proposed changes and squashed them

**Review by DerekBum — APPROVED** [review]
Thanks for the patch!

**Pull request merged**